routingpy/optimization.py

```diff
@@ -0,0 +1,528 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2021 GIS OPS UG
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+from routingpy.utils import decode_polyline5
+
+
+class Job:
+    def __init__(
+        self,
+        id,
+        description=None,
+        location=None,
+        location_index=None,
+        setup=None,
+        service=None,
+        delivery=None,
+        pickup=None,
+        skills=None,
+        priority=None,
+        time_windows=None,
+    ):
+        """
+        :param id: The job id
+        :type id: int
```

Comment:
you probably just took this from the existing stuff. let's deprecate this type of type hinting, at least for new classes/defs. can we pull them into the args list?